java/ql/src/semmle/code/java/dataflow/internal/DataFlowUtil.qll

```diff
@@ -408,29 +409,16 @@ predicate simpleLocalFlowStep(Node node1, Node node2) {
   or
   summaryStep(node1, node2, "value")
   or
-  exists(MethodAccess ma, Method m |
-    ma = node2.asExpr() and
-    m = ma.getMethod() and
-    m.getDeclaringType().hasQualifiedName("java.util", "Objects") and
+  exists(MethodAccess ma, ValuePreservingCallable c, int argNo |
+    ma.getCallee() = c and c.returnsValue(argNo)
+  |
+    node2.asExpr() = ma and
     (
-      m.hasName(["requireNonNull", "requireNonNullElseGet"]) and node1.asExpr() = ma.getArgument(0)
-      or
-      m.hasName("requireNonNullElse") and node1.asExpr() = ma.getAnArgument()
+      node1.asExpr() = ma.getArgument(argNo)
```

Comment:
Instead of using the syntactic `MethodAcces.getArgument(int)` we should use the `ArgumentNode.argumentOf(call, pos)`. That also covers the `-1` case.